zerver/views/upload.py

```diff
@@ -199,7 +199,7 @@ def serve_file(
     is_authorized = validate_attachment_request(maybe_user_profile, path_id, realm)
 
     if is_authorized is None:
-        return HttpResponseNotFound(_("<p>File not found.</p>"))
+        return HttpResponseNotFound(_("<p>This file does not exist or has been deleted.</p>"))
```

Comment:
As noted above, this is the actually most important place to change.  Since it wants the same sort of logic as `zerver.views.thumbnail`, you should consider how you can share that code.